lib/benchmark/bench.go

```diff
@@ -143,7 +186,24 @@ func (tc *TeleportClient) Benchmark(ctx context.Context, bench Benchmark) (*Benc
 			}
 		}
 	}
+}
 
+func produceMeasures(ctx context.Context, rate int, c chan<- *benchMeasure) {
+	interval := time.Duration(float64(1) / float64(rate) * float64(time.Second))
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+
+			measure := &benchMeasure{
+				Start: time.Now(),
+			}
+			c <- measure
```

Comment:
Earlier there was a select here:

```
select {
    case c <- measure:
    case <-ctx.Done():
        return
    }
```

Is that no longer needed?

Reply:
I think there is a latent (not something you introduced) coordinated omission bug in this code.  Either the `select` or the direct send `c <- measure` will block this goroutine if the buffered channel is full -- a.k.a. work is being produced faster than worker threads can execute it.  The producer won't continue to the next iteration of its loop until something  receives off the request channel.  @klizhentas can you double check my thought process here?

Assuming this is an issue, we need to decouple creating the measure from something being ready to work on it.  Brainstorming a couple options:

1)  let goroutines pile up to handle the queue:
```
go func(val int) { c <- val }(i)
```

My gut doesn't  like this, but my rational mind hasn't fully worked out why.

2) Pregenerate all the measures in a big slice (we can calculate their expected start time relative to the start of the benchmark) In this loop, send measures as soon as it is past their foretold start time.  In the worker go routine, record observed start time in addition to completion time, allowing us to measure the difference between service time and response time.  This gives good data, but is kinda hairy.  Also we want to make sure generating all the measures in one big block doesn't throw off any timing

3) Other ideas?

The work here should probably be in a seperate PR, as fixing this latent bug is orthogonal to the changes at hand.

Reply:
> 1. let goroutines pile up to handle the queue:
> 
> ```
> go func(val int) { c <- val }(i)
> ```
>
> My gut doesn't like this, but my rational mind hasn't fully worked out why.

After mulling it over, my rational argument is: Why have a cap on the number of worker goroutine at all if we're going to kick off a goroutine per measure.  If we pursue this option, we might as well tear out `--threads` and related logic, and do the benchmark work in the goroutine the producer spawns -- no need for a request channel at all.

Reply:
This issue is now tracked in https://github.com/gravitational/teleport/issues/4601 and will be handled outside of this PR.